File: lista/Lista Estática/Java/Lista Circular/Listavel.java
public interface Listavel {
	
	//anexa(insere ao final da lista) a lista o novo dado fornecido
	void anexar(Object dado);
	//insere o novo dado fornecido na lista, numa posicao logica informada 
	void inserir(int posicao, Object dado);	
	//retorna o elemento que está numa posicao logica informada
	Object selecionar(int posicao);
	//retorna o elemento que está numa posicao logica informada
	Object[] selecionarTodos();
	//substitui o elemento de uma posicao logica informada,
	//pelo novo elemento fornecido
	void atualizar(int posicao, Object novoDado);	
	//remove o elemento de uma posicao logica informada
	Object apagar(int posicao);
	
	//métodos auxiliares
	boolean estaCheia();
	boolean estaVazia();
	String imprimir();

	//Object apagarFim();
	//Object apagarInicio();
	//void inserirInicio(Object dado);
	//void inserirFim(Object dado); mesmo que void anexar(Object dado);
	//Object obterInicio();
	//Object obterFim();	
	//int indiceDe(Object dado);
	//int ultimoIndiceDe(Object dado);
	//boolean contem(Object dado);
	//Object[] limpar();
	//int tamanho();
}

File: lista/Lista Estática/Kotlin/Lista Circular/Listavel.kt
//https://www.javainuse.com/java2kot
//https://play.kotlinlang.org
interface Listavel {
    // Metodos principais
    fun anexar(dado: Any?)
    fun inserir(posicao: Int, dado: Any?)
    fun selecionar(posicao: Int): Any?
    fun selecionarTodos(): Array<Any?>
    fun atualizar(posicao: Int, dado: Any?)
    fun apagar(posicao: Int): Any?

    // Metodos auxiliares
    fun estaCheia(): Boolean
    fun estaVazia(): Boolean
    fun imprimir(): String
}